Source/JavaScriptCore/bytecode/CallLinkInfo.cpp

```diff
@@ -159,8 +159,6 @@ void CallLinkInfo::setMonomorphicCallee(VM& vm, JSCell* owner, JSObject* callee,
         MacroAssembler::repatchNearCall(static_cast<OptimizingCallLinkInfo*>(this)->m_callLocation, CodeLocationLabel<JSEntryPtrTag>(codePtr));
         MacroAssembler::repatchPointer(u.codeIC.m_codeBlockLocation, codeBlock);
         MacroAssembler::repatchPointer(u.codeIC.m_calleeLocation, callee);
-#else
-        RELEASE_ASSERT_NOT_REACHED();
```

Comment:
Why are we removing these asserts? Wouldn't using `!isDataIC()` when the JIT is disabled be wrong?

Reply:
Added.